sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTableScanExec.scala

```diff
@@ -0,0 +1,501 @@
+                                hadoopConf)
+  }
+
+  private def createReadRDDForPartitions(
+                readFile: PartitionedFile => Iterator[InternalRow],
+                hiveTableRelation: HiveTableRelation,
+                readSchema: StructType,
+                sparkSession: SparkSession,
+                hadoopConf: Configuration
+              ): RDD[InternalRow] = {
+    val partitionColTypes = hiveTableRelation.partitionCols.map(_.dataType)
+    val dirsWithPartValues = prunedPartitions.map { p =>
+      // No need to check if partition directory exists.
+      // FileSystem.listStatus() handles this for GpuHiveTableScanExec,
+      // just like for Apache Spark.
+      val uri = p.getDataLocation.toUri
+      val partValues: Seq[Any] = {
+        p.getValues.asScala.zip(partitionColTypes).map {
+          case (value, dataType) => castFromString(value, dataType)
+        }
+      }
+      val partValuesAsInternalRow = InternalRow.fromSeq(partValues)
+
+      (uri, partValuesAsInternalRow)
+    }
+
+    createReadRDDForDirectories(readFile,
+      dirsWithPartValues, readSchema, sparkSession, hadoopConf)
+  }
+
+  lazy val inputRDD: RDD[InternalRow] = {
+    // Assume Delimited text.
+    // Note: The populated `options` aren't strictly required for text, currently.
+    //       These are added in case they are required for table read in the future,
+    //       (like with Hive ORC tables).
+    val options                   = hiveTableRelation.tableMeta.properties ++
+                                    hiveTableRelation.tableMeta.storage.properties
+    val hadoopConf                = sparkSession.sessionState.newHadoopConfWithOptions(options)
+    val broadcastHadoopConf       = sparkSession.sparkContext.broadcast(
+                                      new SerializableConfiguration(hadoopConf))
+    val sqlConf                   = sparkSession.sessionState.conf
+    val rapidsConf                = new RapidsConf(sqlConf)
+    val requestedOutputDataSchema = getRequestedOutputDataSchema(hiveTableRelation.tableMeta.schema,
+                                                                 partitionAttributes,
+                                                                 requestedAttributes)
+    val reader                    = buildReader(sqlConf,
+                                                broadcastHadoopConf,
+                                                rapidsConf,
+                                                hiveTableRelation.tableMeta.dataSchema,
+                                                hiveTableRelation.tableMeta.partitionSchema,
+                                                requestedOutputDataSchema,
+                                                options)
+    val rdd = if (hiveTableRelation.isPartitioned) {
+      createReadRDDForPartitions(reader, hiveTableRelation, requestedOutputDataSchema,
+                                 sparkSession, hadoopConf)
+    }
```

Comment:
nit same here the format is off from the regular format.

Reply:
I've fixed the formatting now.